### ISSUE TITLE
[Bug]: Timeout when taking screenshot in v1.16.1

### ISSUE THREAD

**Issue by ikesnowy** [microsoft/playwright-dotnet#1818]
### Playwright version

1.16.1

### Operating system

Linux

### What browsers are you seeing the problem on?

Chromium

### Other information

using the docker image mcr.microsoft.com/playwright:v1.16.1 and install dotnet runtime manually.

### What happened? / Describe the bug

experiencing timeout exception when taking screenshot after upgrading from v1.15.4  to v1.16.1

repro: https://github.com/ikesnowy/playwright-repro

### Code snippet to reproduce your bug

```shell
using System.Threading.Tasks;
using Microsoft.Playwright;

namespace PlaywrightRepro
{
    class Program
    {
        public static async Task Main()
        {
            using var playwright = await Playwright.CreateAsync();
            await using var browser = await playwright.Chromium.LaunchAsync(new() { Headless = true });
            var page = await browser.NewPageAsync();
            await page.GotoAsync("https://playwright.dev/dotnet");
            await page.ScreenshotAsync(new() { Path = "screenshot.png" });
        }
    }

}
```

### Relevant log output

```shell
2021-11-03T04:30:43.540Z pw:api navigating to "https://playwright.dev/dotnet", waiting until "load"
2021-11-03T04:30:45.479393661Z 2021-11-03T04:30:45.478Z pw:api   navigated to "https://playwright.dev/dotnet/"
2021-11-03T04:30:46.271335798Z 2021-11-03T04:30:46.270Z pw:api   "domcontentloaded" event fired
2021-11-03T04:30:46.567798279Z 2021-11-03T04:30:46.567Z pw:api   navigated to "https://ghbtns.com/github-btn.html?user=microsoft&repo=playwright-dotnet&type=star&count=true&size=large"
2021-11-03T04:30:46.571152271Z 2021-11-03T04:30:46.570Z pw:api   "domcontentloaded" event fired
2021-11-03T04:30:53.668471295Z 2021-11-03T04:30:53.667Z pw:api   "load" event fired
2021-11-03T04:30:54.134352394Z 2021-11-03T04:30:54.133Z pw:api   "networkidle" event fired
2021-11-03T04:31:23.790628895Z Unhandled exception. System.TimeoutException: Timeout 30000ms exceeded.
2021-11-03T04:31:23.790663178Z    at Microsoft.Playwright.Transport.Connection.SendMessageToServerAsync[T](String guid, String method, Object args)
2021-11-03T04:31:23.790667746Z    at Microsoft.Playwright.Transport.Channels.PageChannel.ScreenshotAsync(String path, Nullable`1 fullPage, Clip clip, Nullable`1 omitBackground, Nullable`1 type, Nullable`1 quality, Nullable`1 timeout)
2021-11-03T04:31:23.790670916Z    at Microsoft.Playwright.Core.Page.ScreenshotAsync(PageScreenshotOptions options)
2021-11-03T04:31:23.790674620Z    at PlaywrightRepro.Program.Main() in /src/Program.cs:line 14
2021-11-03T04:31:23.790678273Z    at PlaywrightRepro.Program.Main() in /src/Program.cs:line 14
2021-11-03T04:31:23.790681163Z    at PlaywrightRepro.Program.<Main>()
```

**Comment by cwwarren** [comment]
We discovered this today as well with a small microservice that's very similar to the given example with a HTTP API over it.

It seems intermittent and unreliable to reproduce -- although it happened in more than 50% of the times we tested. When two replicas of the service are provided the same input HTML we observed either could hang at 100% CPU while the other rendered the page fine. Once the bug is encountered the container was stuck at 100% CPU until killed.
